.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+permissions:
+  contents: read
```

Comment:
Is this required ? or just nice to have?

Reply:
So I grabbed that from a template github action for running Django tests. it might not be needed, but i haven't tested taking it out.

Reply:
Actually, this is a good practice, I believe. 

It seems that this `permissions` section serves the purpose of customizing the default permissions for the `GITHUB_TOKEN`  according to the [docs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs)

That said, the default permissions of the `GITHUB_TOKEN` have some [restrictive enough permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) but ultimately they depend on the org's settings. So I think enforcing them to `read` is not a bad thing.